### PR TITLE
Remove expectThrowsAsync

### PR DESCRIPTION
**Pull request by Milena-Czierlinski**
# Readiness checklist

-   [x] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

File: packages/transport/test/testHelpers/TestUtil.ts [hunk 1]
```diff
@@ -135,11 +135,7 @@ export class TestUtil {
         });
     }
 
-    public static async expectThrowsAsync(method: Function | Promise<any>, customExceptionMatcher: (e: Error) => void): Promise<void>;
-
-    public static async expectThrowsAsync(method: Function | Promise<any>, errorMessageRegexp: RegExp | string): Promise<void>;
-
-    public static async expectThrowsAsync(method: Function | Promise<any>, errorMessageRegexp: RegExp | string | ((e: Error) => void)): Promise<void> {
+    private static async expectThrowsAsync(method: Function | Promise<any>, errorMessageRegexp: RegExp | string | ((e: Error) => void)): Promise<void> {
```
Comment (jkoenig134) [inline packages/transport/test/testHelpers/TestUtil.ts:138]: why is this expectThrowsAsync not removed?

Comment (Milena-Czierlinski) [reply packages/transport/test/testHelpers/TestUtil.ts:138]: Because it is used in `expectThrowsRequestErrorAsync` to check the correct error code.

Comment (jkoenig134) [reply packages/transport/test/testHelpers/TestUtil.ts:138]: We could use .rejects there too, or not? Or at least move the logic inside the other function.

Comment (jkoenig134) [reply packages/transport/test/testHelpers/TestUtil.ts:138]: I just saw you made it async ... then it's okayish.

**Review by jkoenig134 — DISMISSED** [review]

File: packages/transport/test/testHelpers/TestUtil.ts [hunk 2]
```diff
@@ -121,44 +121,23 @@ export class TestUtil {
...
-        return await this.expectThrowsAsync(method, (error: Error) => {
-            if (errorMessageRegexp) {
-                expect(error.message).toMatch(new RegExp(errorMessageRegexp));
-            }
+    public static async expectThrowsRequestErrorAsync(method: Promise<any>, errorMessageRegexp?: RegExp | string, status?: number): Promise<void> {
+        await expect(method).rejects.toThrow(RequestError);
 
+        try {
+            await method;
+        } catch (error) {
```
[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/transport/test/testHelpers/TestUtil.ts:129]:
```
    public static async expectThrowsRequestErrorAsync(method: Promise<any>, errorMessageRegexp?: RegExp | string, status?: number): Promise<void> {
        let error: Error | undefined;

        try {
            await method;
        } catch (e) {
            error = e as Error;
        }

        expect(error).toBeInstanceOf(RequestError);

        const requestError = error as RequestError;

        if (errorMessageRegexp) {
            expect(requestError.message).toMatch(new RegExp(errorMessageRegexp));
        }

        if (status) {
            expect(requestError.status).toStrictEqual(status);
        }
    }
```

i suggest this

[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/transport/test/testHelpers/TestUtil.ts:129]: and `method` should be renamed to `promise`

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**